conda/cli/install.py

```diff
@@ -315,82 +327,64 @@ def ensure_update_specs_exist(prefix: str, specs: list[str]):
             raise PackageNotInstalledError(prefix, spec.name)
 
 
-def install_clone(args, parser):
-    """Executes an install of a new conda environment by cloning."""
-    prefix = context.target_prefix
-    index_args = get_index_args(args)
-
-    # common validations for all types of installs
-    validate_install_command(prefix=prefix, command="create")
+def _assemble_environment(
+    name: str | None = None,
+    prefix: str | None = None,
+    specs: Iterable[str] = (),
+    files: Iterable[str] = (),
+    inject_default_packages: bool = True,
+) -> Environment:
+    # First, let's create an 'Environment' for the information exposed in the CLI (no files)
+    if inject_default_packages:
+        names = {MatchSpec(pkg).name for pkg in specs}
+        for pkg in context.create_default_packages:
+            pkg_name = MatchSpec(pkg).name
+            if pkg_name not in names:
+                specs.append(pkg)
+
+    requested_packages = []
+    fetch_explicit_packages = []
 
-    clone(
-        args.clone,
-        prefix,
-        json=context.json,
-        quiet=context.quiet,
-        index_args=index_args,
+    for spec in specs:
+        if is_package_file(spec):
+            fetch_explicit_packages.append(spec)
+        else:
+            requested_packages.append(common.arg2matchspec(spec))
+
+    # transform explicit packages into package records
+    explicit_packages = []
+    if fetch_explicit_packages:
+        explicit_packages = get_package_records_from_explicit(fetch_explicit_packages)
```

Comment:
This function has a potential side-effect of pre-downloading artifacts, only to build a transaction that is then presented to the user to be accepted or rejected with a prompt.

Is there a chance we can create `PackageRecord` shims with just the info provided in the URL (should be enough: channel, subdir, package name, version, build) so we can assemble a good enough Transaction to ask for confirmation? Then, once the user has chosen "yes" in the prompt, we do perform the downloads and create the actual transaction object from the Package Cache info.